### DISCUSSION TITLE
Feature Request: Add Editable Code Snippets to Documentation

### DISCUSSION THREAD

**Discussion by Farooq-Abdulla** [mantinedev/mantine#6430] [Feature requests]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.3.0

### What package has an issue?

@mantine/core

### What framework do you use?

Next.js

### In which browsers you can reproduce the issue?

All

#### Description:
It would be beneficial to have interactive and editable code snippets in the Mantine documentation. This would allow developers to experiment with component usage directly within the documentation, enhancing the learning experience and speeding up the development process.

#### Benefits:
- **Interactive Learning**: Users can modify the code snippets and see the results immediately, which helps in better understanding how the components work.
- **Quick Prototyping**: Developers can quickly test different configurations and properties without leaving the documentation.
- **Enhanced Engagement**: Interactive documentation is more engaging and user-friendly.

#### Example:
Here is an example of how this feature might look:
- A code snippet that can be edited in a live code editor embedded in the documentation page.
- Real-time preview of the changes made in the code snippet.

<img width="803" alt="Screenshot 2024-06-20 at 6 21 47 PM" src="https://github.com/mantinedev/mantine/assets/145077556/2436b8de-5442-4d57-bf16-a344381a7ecd">

Thank you for considering this feature request.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by Farooq-Abdulla** [comment]
I believe this enhancement will significantly improve the developer experience by making the documentation more interactive and engaging. It will also facilitate quicker prototyping and learning for new users of Mantine.

**Comment by rtivital** [comment]
It is not planned to be implemented. It requires loading additional dependencies to the website that will make it much slower considering that browser cache on gh-pages does not work well.
